src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2297,6 +2300,83 @@ void os::Linux::print_container_info(outputStream* st) {
   st->cr();
 }
 
+// SapMachine 2019-06-26: 8225345: Provide Cloud IAAS related info on Linux in the hs_err file
+static int check_matching_lines_from_file(const char* filename, const char* keywords_to_match[]) {
+  char line[500];
+  FILE* fp = fopen(filename, "r");
+  if (fp == NULL) {
+    return -1;
+  }
+
+  while (fgets(line, sizeof(line), fp) != NULL) {
+    int i = 0;
+    while (keywords_to_match[i] != NULL && line != NULL) {
+      if (strstr(line, keywords_to_match[i]) != NULL) {
+        fclose(fp);
+        return i;
+      }
+      i++;
+    }
+  }
+  fclose(fp);
+  return -1;
+}
+
+// SapMachine 2019-06-26: 8225345: Provide Cloud IAAS related info on Linux in the hs_err file
+// add Cloud IAAS information where possible, a basic detection can be done by using dmi info
+// Google GCP: /sys/class/dmi/id/product_name contains 'Google Compute Engine' (or just 'Google')
+// Alibaba   : /sys/class/dmi/id/product_name contains 'Alibaba Cloud ECS'
+// Azure     : /sys/class/dmi/id/chassis_asset_tag contains '7783-7084-3265-9085-8269-3286-77'
+//             (ASCII-encoded: 'MS AZURE VM')
+// AWS KVM/Baremetal : /sys/class/dmi/id/chassis_asset_tag contains 'Amazon EC2'
+// AWS Xen           : /sys/class/dmi/id/bios_version  and /sys/class/dmi/id/product_version
+//                     contain amazon (plus some more info)
+//                     /sys/class/dmi/id/bios_vendor and /sys/class/dmi/id/chassis_vendor contain 'Xen'
+void os::Linux::print_cloud_info(outputStream* st) {
+  // dmidir is /sys/class/dmi/id
+  const char* filename = "/sys/class/dmi/id/product_name";
+  const char* kwcld[] = { "Google", "Google Compute Engine", "Alibaba Cloud", NULL };
+  int res = check_matching_lines_from_file(filename, kwcld);
+  if (res != -1) { // a matching Cloud identifier has been found
+    st->print_cr("Cloud infrastructure found:");
+    if (res == 0 || res == 1) {
+      st->print_cr("Google cloud detected");
+    }
+    if (res == 2) {
+      st->print_cr("Alibaba cloud detected");
+    }
+  }
```

Comment:
Would it make sense to return from this function when one cloud has been detected?
Otherwise it reads four files in best case and five files in worst case.

Reply:
yes  I do that already in my latest  version (in SAPJVM).